### PR TITLE
Update dependencies to fix deprecation warnings

### PR DESCRIPTION
**Pull request by lydell**
Fixes the following deprecations:

    npm WARN deprecated request@2.79.0: request has been deprecated, see https://github.com/request/request/issues/3142
    npm WARN deprecated hawk@3.1.3: This module moved to @hapi/hawk. Please make sure to switch over as this distribution is no longer supported and may contain bugs and critical security issues.
    npm WARN deprecated cryptiles@2.0.5: This version has been deprecated in accordance with the hapi support policy (hapi.im/support). Please upgrade to the latest version to get the best features, bug fixes, and security patches. If you are unable to upgrade at this time, paid support is available for older versions (hapi.im/commercial).
    npm WARN deprecated hoek@2.16.3: This version has been deprecated in accordance with the hapi support policy (hapi.im/support). Please upgrade to the latest version to get the best features, bug fixes, and security patches. If you are unable to upgrade at this time, paid support is available for older versions (hapi.im/commercial).
    npm WARN deprecated boom@2.10.1: This version has been deprecated in accordance with the hapi support policy (hapi.im/support). Please upgrade to the latest version to get the best features, bug fixes, and security patches. If you are unable to upgrade at this time, paid support is available for older versions (hapi.im/commercial).
    npm WARN deprecated sntp@1.0.9: This module moved to @hapi/sntp. Please make sure to switch over as this distribution is no longer supported and may contain bugs and critical security issues.
    npm WARN deprecated core-js@2.6.11: core-js@<3 is no longer maintained and not recommended for usage due to the number of issues. Please, upgrade your dependencies to the actual version of core-js@3.

And some vulnerabilities:

    found 58 vulnerabilities (26 low, 7 moderate, 24 high, 1 critical)
      run `npm audit fix` to fix them, or `npm audit` for details

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,8 +1,5 @@
 language: node_js
 node_js:
-  - 0.10
-  - 0.12
-  - 4
-  - 6
-  - 8
-  - 9
```
Comment (thlorenz) [inline .travis.yml:8]: can we keep support down to Node.js 8 at least?

File: package.json [hunk 2]
```diff
@@ -10,21 +10,21 @@
...
-    "minimist": "~1.2.0",
-    "underscore": "~1.8.3",
-    "update-section": "^0.3.0"
+    "@textlint/markdown-to-ast": "^6.1.7",
+    "anchor-markdown-header": "^0.5.7",
+    "htmlparser2": "^4.1.0",
+    "minimist": "^1.2.5",
+    "underscore": "^1.10.2",
+    "update-section": "^0.3.3"
   },
```
Comment (thlorenz) [inline package.json:25]: Upgrade looks good, but please keep using `~` for ranges to limit issues introduced by dependency updates.

File: package.json [hunk 3]
```diff
@@ -10,21 +10,21 @@
...
   "devDependencies": {
-    "tap": "~10.2.1"
```
Comment (thlorenz) [inline package.json:27]: For the test runner `^` is fine.

File: .travis.yml (hunk 1 above)
Comment (lydell) [reply .travis.yml:8]: Oh, I thought Node.js 8 was EOL?

Comment (AndrewSouthpaw) [reply .travis.yml]: This was addressed in a recent PR, calling this resolved.

**Comment by AndrewSouthpaw** [comment]
@thlorenz I think this is ready to go, if you want to give the final approval since you requested changes?

**Comment by AndrewSouthpaw** [comment]
Calling this good to go.

**Pull request merged**